### PR TITLE
docs: add Apache APISIX-Datadog plugin doc

### PR DESCRIPTION
**Pull request by yzeng25**
Add a doc for Apache APISIX's Datadog plugin.

### CODE REVIEW

**Review by yzeng25 — COMMENTED** [review]
Does anyone knows how to add images here? I did not see any folders with images inside...

File: apache-apisix/collecting-eks-cluster-metrics.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+## How to Use Datadog with Apache APISIX
+
+### Activate the APISIX-Datadog plugin
+
+The following is an example on how to activate the datadog plugin for a specific route. We are assumming your `dogstatsd` agent is already up an running.
+
+```shell
+# enable plugin for a specific route
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (apigirl) [inline apache-apisix/collecting-eks-cluster-metrics.md:92]: Same comment as https://github.com/DataDog/documentation/pull/12234/files#r747878989

Comment (yzeng25) [reply apache-apisix/collecting-eks-cluster-metrics.md:92]: Yep, these aren't real keys.

**Comment by apigirl** [comment]
Hi @yzeng25 - we've added a separate PR [here](https://github.com/DataDog/documentation/pull/12345) to update this as per @jeremy-lqs email response to your original request.

**Comment by yzeng25** [comment]
> Hi @yzeng25 - we've added a separate PR [here](https://github.com/DataDog/documentation/pull/12345) to update this as per @jeremy-lqs email response to your original request.

Hi @apigirl , 

I left a comment in https://github.com/DataDog/documentation/pull/12345. There is a typo in the "Language" column. Other than that, everything else LGTM. Thanks for doing this!

**Review by moonming — COMMENTED** [review]
nice work 👍

**Comment by yzeng25** [comment]
As discussed with @jeremy-lq in emails, all documents in this blog shall be written and maintained by Datadog. This pull request is replaced by DataDog/documentation#12345. 

CC @apigirl @moonming

**Pull request closed without merging**